cms/djangoapps/contentstore/views/component.py

```diff
@@ -293,6 +294,8 @@ def create_template_dict(name, cat, boilerplate_name=None, is_common=False):
     course_advanced_keys = course.advanced_modules
     advanced_component_templates = {"type": "advanced", "templates": [], "display_name": _("Advanced")}
     # Set component types according to course policy file
+    course_advanced_keys = course_advanced_keys or []
+    course_advanced_keys = list(set(course_advanced_keys + XBLOCKS_ALWAYS_IN_STUDIO))
```

Comment:
Ideally, I would just store `course_advanced_keys` as a `set` (instead of recasting to a `list`, but the following `if` checks for strict-typing instead of duck-typing and I wanted to keep this diff small.
```python
if isinstance(course_advanced_keys, list):
    # ...
```
vs.
```python
if hasattr(course_advanced_keys, '__iter__'):
    # ...
```